networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -1935,3 +1939,200 @@ def dist_path(v):
 
     return {v: dist_path(v) for v in G}
 
+def orienteering(G, init_sol, starting, ending, scores, temp=100,
```

Comment:
These variable names are a little unclear.

`starting` and `ending` should be `source` and `target` to match the other NetworkX shortest path functions.

`inner_loop` and `outer_loop` don't give any information about what effect they have. Maybe something like `num_neighbors_considered`? I'm not sure.